pkg/handlers/http-tracer.go

```diff
@@ -109,92 +113,73 @@ func (r *recordResponseWriter) Flush() {
 	r.ResponseWriter.(http.Flusher).Flush()
 }
 
-// Return response headers.
-func (r *recordResponseWriter) Headers() []byte {
-	return r.headers.Bytes()
-}
-
 // Return response body.
 func (r *recordResponseWriter) Body() []byte {
 	return r.body.Bytes()
 }
 
-// TraceReqHandlerFunc logs request/response headers and body.
-func TraceReqHandlerFunc(f http.HandlerFunc, output io.Writer, logBody bool) http.HandlerFunc {
+// Trace gets trace of http request
+func Trace(f http.HandlerFunc, logBody, logNodeName bool, w http.ResponseWriter, r *http.Request) (trace.Info, error) {
+
 	name := runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
 	name = strings.TrimPrefix(name, "github.com/minio/minio/cmd.")
 	bodyPlaceHolder := []byte("<BODY>")
-
-	return func(w http.ResponseWriter, r *http.Request) {
-		const timeFormat = "2006-01-02 15:04:05 -0700"
-		var reqBodyRecorder *recordRequest
-
-		// Generate short random request ID
-		reqID := fmt.Sprintf("%f", float64(time.Now().UnixNano())/1e10)
-
-		reqBodyRecorder = &recordRequest{Reader: r.Body, logBody: logBody}
-		r.Body = ioutil.NopCloser(reqBodyRecorder)
-
-		// Setup a http response body recorder
-		respBodyRecorder := &recordResponseWriter{ResponseWriter: w, logBody: logBody}
-
-		b := bytes.NewBuffer(nil)
-		fmt.Fprintf(b, "[REQUEST %s] [%s] [%s]\n", name, reqID, time.Now().Format(timeFormat))
-
-		f(respBodyRecorder, r)
-
-		// Build request log and write it to log file
-		fmt.Fprintf(b, "%s %s", r.Method, r.URL.Path)
-		if r.URL.RawQuery != "" {
-			fmt.Fprintf(b, "?%s", r.URL.RawQuery)
-		}
-		fmt.Fprintf(b, "\n")
-
-		fmt.Fprintf(b, "Host: %s\n", r.Host)
-		for k, v := range r.Header {
-			fmt.Fprintf(b, "%s: %s\n", k, v[0])
+	var reqBodyRecorder *recordRequest
+
+	// Generate short random request ID
+	reqID := fmt.Sprintf("%f", float64(time.Now().UnixNano())/1e10)
```

Comment:
reqID can be removed as we always print request/response in one shot (i.e there is no intermingling of requests and responses)